samcli/lib/providers/provider.py

```diff
@@ -29,6 +29,8 @@ class Function(NamedTuple):
     Named Tuple to representing the properties of a Lambda Function
     """
 
+    # Function id, can be Logical ID or any function identifier to define a function in specific IaC
```

Comment:
What about the function name? `name` can be the Function Name or the logical id (reading the comment below)

Reply:
Looks like the `name` is function name is always logical ID. When the `FunctionName` is not defined, the function name is the same as logical id. I updated that comment to make it more clear.
https://github.com/aws/aws-sam-cli/blob/ec9f5d89971b823fa01071c96b7de90ea5e5303a/samcli/lib/providers/sam_function_provider.py#L340

https://github.com/aws/aws-sam-cli/blob/ec9f5d89971b823fa01071c96b7de90ea5e5303a/samcli/lib/providers/sam_function_provider.py#L129